line_robots/canvas.cpp

```diff
@@ -60,12 +62,20 @@ void Canvas::dropEvent(QGraphicsSceneDragDropEvent *event)
                     event->ignore();
                     errorMsg(4);
                 } else {
+                    setSpeedBox();
+                    Robot *newRobot = new Robot(x, y, lineDroppedOn, dragObjectType);
+                    newRobot->setSpeed(roboticSpeed);
+                    newRobot->setZValue(1);
+                    qDebug()<<lineDroppedOn->line();
+                    qDebug()<<newRobot->pos();
```

Comment:
These lines should be deleted - handled in the if statement which prevents a robot from being created if the user cancels out of the pop up asking for robot speed and color.